dali/image/generic_image.cc

```diff
@@ -27,6 +27,8 @@ std::pair<std::shared_ptr<uint8_t>, Image::Shape>
 GenericImage::DecodeImpl(DALIImageType image_type,
                          const uint8_t *encoded_buffer,
                          size_t length) const {
+  DALI_ENFORCE(image_type != DALI_ANY_DATA, "Host decoder doesn't support ANY_DATA image type for "
+                                            "this image");
```

Comment:
Why?

Reply:
You can pass `IMREAD_ANYCOLOR` and/or `IMREAD_UNCHANGED` to imdecode to keep the original number of channels.

Reply:
IMREAD_ANYCOLOR will preserve whatever color format was there to begin with. It would take some trial and error to see what we'd get out of ordinary JPEGs, for example - would it be BGR or RGB or maybe YCbCr? I'll poke it a bit.

Reply:
This PR is to make sure we are at least consistent and fail instead of providing random results or out-of-bound memory access.
Supporting multichannel image formats is a full-fledged feature, not a quick fix.

Reply:
I don't know why it would result in out-of-bounds access. Just throwing away support for alpha in PNGs (which OpenCV _does_ support) seems quite wasteful.